### PR TITLE
Fix ios cache

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
We were caching the token to not access too often the Keychain on iOS, but that cache was not getting invalidated on logout.

This was creating problems with Push Notifications where a wrong token was used to get the notification message, resulting in the message showing "session expired".

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-52991

#### QA Steps
- Log in
- Close the app
- Receive a notification: It shows the message alright
- Log out
- Log back in
- Close the app
- Receive a notification

Expected:
The notification shows alright

Before the fix:
The notfication shows as if the message was "session expired"

#### Release Note
```release-note
Fix case where notifications would show "Session expired" instead of the message.
```

### CODE REVIEW

File: app/managers/session_manager.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import FastImage from 'react-native-fast-image';
 
 import {storeOnboardingViewedValue} from '@actions/app/global';
 import {cancelSessionNotification, logout, scheduleSessionNotification} from '@actions/remote/session';
+import {invalidateKeychainCache} from '@app/utils/mattermost_managed';
```
Comment (enahum) [inline app/managers/session_manager.ts:10]:
```suggestion
import {invalidateKeychainCache} from '@utils/mattermost_managed';
```

**Review by enahum — APPROVED** [review]
one small change on the import

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍

**Comment by harshilsharma63** [comment]
@larkox do we not have to do anything similar for Android?

**Comment by enahum** [comment]
> @larkox do we not have to do anything similar for Android?

No, this is an iOS issue only

**Comment by enahum** [comment]
@yasserfaraazkhan please review asap

**Review by yasserfaraazkhan — APPROVED** [review]
thank you @larkox 
I've tried multiple times logging out and verifying the push notification. Did not see 'session expired' message

4132E0A4-BABD-488B-8D6E-D0920DF8DFC4

B945F8EB-6D68-498E-A073-777C20344FE6

**Comment by EduardoSellanes** [comment]
I know this is not the right place to comment/report (sorry for that), but I don't have access to comment in the ticket, I've been checking this issue on production 2.6 version, and it's still present... 
please check the attached screen recording, is there any way to reopen the ticket?
https://github.com/mattermost/mattermost-mobile/assets/5940718/c22e02f3-27ea-4ef3-b3f8-a48feb995622
thanks!

**Comment by EduardoSellanes** [comment]
It seems that NotificationService is getting the token from its cache instance

please, take a look to the debug

```
GET A PUSHNOTIFICATION
14:30:18.610728-0400    NotificationService    getToken => return token dwu8of4jg7bozfx5gugjbqdbqy
14:30:18.610768-0400    NotificationService               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy
14:30:18.817892-0400    NotificationService    getToken => return CACHE dwu8of4jg7bozfx5gugjbqdbqy
14:30:18.817956-0400    NotificationService               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy

14:30:19.890921-0400    Mattermost    getToken => return token dwu8of4jg7bozfx5gugjbqdbqy
14:30:19.891002-0400    Mattermost               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy
14:30:19.892566-0400    Mattermost    getToken => return CACHE dwu8of4jg7bozfx5gugjbqdbqy
14:30:19.892657-0400    Mattermost               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy
14:30:19.896557-0400    Mattermost    getToken => return CACHE dwu8of4jg7bozfx5gugjbqdbqy
14:30:19.896684-0400    Mattermost               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy

LOG OUT FROM THE ACCOUNT
14:32:36.726115-0400    Mattermost    invalidateToken => CACHE removed!

LOG IN WITH THE SAME ACCOUNT
GET A NEW PUSHNOTIFICATION
14:33:43.850184-0400    NotificationService    getToken => return CACHE dwu8of4jg7bozfx5gugjbqdbqy
14:33:43.850220-0400    NotificationService               buildURLRequest got token dwu8of4jg7bozfx5gugjbqdbqy

14:33:44.587018-0400    Mattermost    getToken => return token odnbt6m5ei887xfzji8ms5i4qa
14:33:44.587117-0400    Mattermost               buildURLRequest got token odnbt6m5ei887xfzji8ms5i4qa
14:33:44.588118-0400    Mattermost    getToken => return CACHE odnbt6m5ei887xfzji8ms5i4qa
14:33:44.588165-0400    Mattermost               buildURLRequest got token odnbt6m5ei887xfzji8ms5i4qa
14:33:44.597541-0400    Mattermost    getToken => return CACHE odnbt6m5ei887xfzji8ms5i4qa
14:33:44.597686-0400    Mattermost               buildURLRequest got token odnbt6m5ei887xfzji8ms5i4qa
```

**Comment by larkox** [comment]
Thank you for the exploration!

We have to communicate with the NotificationService that it has to invalidate the cache. Not sure if there is any native communication route, but worst case scenario, we can use the disk, via the database or just a file.

Another solution would be to not use caching at all.

**Comment by enahum** [comment]
I think is fine to remove the cache for the extension

**Comment by EduardoSellanes** [comment]
ok thanks, was it to resolve a reported performance issue?

**Pull request merged**